### PR TITLE
Check for newer releases

### LINKED ISSUE #124: Check for newer releases on GIthub
It is a priority for sedge users, to run an updated node and an updated version of sedge.

When the app starts, we should check that we don't have a new release available on GitHub, and if that is the case, tell the user that they need to upgrade the app.

We need to embed on the app the version running at compilation time, and check that version against GitHub.

### PR DESCRIPTION
**Pull request by stdevMac**
Resolves #124 

## Changes:
- Add check for newer releases on GitHub

## Types of changes
- [x] New feature (non-breaking change which adds functionality)

## Testing
**Requires testing**
- [x] Yes

**In case you checked yes, did you write tests?**
- [x] Yes

### CODE REVIEW

File: cli/root.go [hunk 1]
```diff
@@ -17,6 +17,7 @@ package cli
 
 import (
 	"fmt"
+	"github.com/NethermindEth/sedge/internal/utils"
 	"os"
 	"strings"
```
Comment (AntiD2ta) [inline cli/root.go:23]:
```suggestion
import (
	"fmt"
	"os"
	"strings"

	"github.com/NethermindEth/sedge/internal/utils"
```

File: cli/version.go [hunk 2]
```diff
@@ -17,6 +17,7 @@ package cli
 
 import (
 	"fmt"
+	"github.com/NethermindEth/sedge/internal/utils"
 
 	"github.com/spf13/cobra"
```
Comment (AntiD2ta) [inline cli/version.go:22]:
```suggestion
	"fmt"
	
	"github.com/NethermindEth/sedge/internal/utils"
	"github.com/spf13/cobra"
```

File: internal/utils/version.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package utils
+
+import (
+	"context"
+	"errors"
+	"github.com/google/go-github/v47/github"
+	"os/exec"
```
Comment (AntiD2ta) [inline internal/utils/version.go:22]:
```suggestion
	"context"
	"errors"
	"os/exec"	
	
	"github.com/google/go-github/v47/github"
```

File: internal/utils/version.go [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+	// owner is the owner of the repository
+	owner = "NethermindEth"
+	// repo is the name of the repository
+	repo = "sedge"
+	// MsgNeedVersionUpdate is the message to display when a new Version is available
+	MsgUnableToCheckVersion = "Unable to check for new Version. Please check manually at " +
+		"https://github.com/NethermindEth/sedge/releases, with error:"
+	MsgNeedVersionUpdate = "A new Version of sedge is available. Please update to the latest Version. See " +
+		"https://github.com/NethermindEth/sedge/releases for more information. Latest detected tag:"
+	MsgVersionUpdated = "You are running the latest version of sedge. Version: "
```
Comment (AntiD2ta) [inline internal/utils/version.go:35]: Currently we are maintaining all of the constant messages from all the packages in `configs/messages.go`. We can discuss in another PR what would be the best approach for messages as this would be out of this PR's scope. For now please move these messages to `configs/messages.go`. Same with error messages.

File: internal/utils/version.go [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+)
+
+var ErrorNoTag = errors.New("no tag found on Github")
+var ErrorCheckingVersion = errors.New("error while checking for new Version, please check your internet connection")
+
+var Version string
+
+func init() {
```
Comment (AntiD2ta) [inline internal/utils/version.go:43]: Please add a summary or description to the functions in this package. For now the only functions without descriptions are the ones from the `cli` package. Check out other functions or methods from other packages to see an example.

Comment (AntiD2ta) [reply internal/utils/version.go:43]: the init function doesn't need a description though. This is for the rest

File: internal/utils/version_test.go [hunk 6]
```diff
@@ -0,0 +1,59 @@
+package utils
+
+import (
+	"github.com/jarcoal/httpmock"
+	"testing"
+)
+
+func TestVersion(t *testing.T) {
```
Comment (AntiD2ta) [inline internal/utils/version_test.go:8]: This unit test is well designed, but the tests cases are not sufficient. It would be good to add test cases for the cases where there is no error and also with a non-empty `tag`

File: internal/utils/version.go [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+			Version = "v0.0.0"
+		} else {
+			Version = version
+		}
+	}
+}
+
+func versionFromGit() (string, error) {
+	cmd := "git tag | sort | tail -n 1"
+	out, err := exec.Command("bash", "-c", cmd).Output()
```
Comment (AntiD2ta) [inline internal/utils/version.go:56]: Please use the code and interfaces designed for running commands in the`internal/pkg/commands` package. Check examples in the `cli` command code to see how to use it. You would need to define a method for the`UnixCMDRunner` struct that executes the command you want. This is designed in a way is easier to maintain a cross platform code. As you probably have realized, this version check doesn't work on Windows right now.

**Review by AntiD2ta — CHANGES_REQUESTED** [review]
Very cool feature, you did a great work here. Just minor changes related to this being your first big contribution.

File: internal/utils/version.go (hunk 7 above)
Comment (stdevMac) [reply internal/utils/version.go]: Please review new changes

File: internal/utils/version_test.go (hunk 6 above)
Comment (stdevMac) [reply internal/utils/version_test.go:9]: Nice, just added one more, can you tell me if that is enough or should I add more?

File: internal/utils/version.go (hunk 5 above)
Comment (stdevMac) [reply internal/utils/version.go:41]: Let me know if the current comments are good

File: internal/utils/version.go (hunk 4 above)
Comment (stdevMac) [reply internal/utils/version.go]: Done

File: internal/utils/version.go (hunk 3 above)
Comment (stdevMac) [reply internal/utils/version.go]: Formatted

File: cli/root.go (hunk 1 above)
Comment (stdevMac) [reply cli/root.go]: Formatted

**Review by AntiD2ta — APPROVED** [review]
